wcomponents-core/src/main/java/com/github/bordertech/wcomponents/util/FileUtil.java

```diff
@@ -0,0 +1,111 @@
+package com.github.bordertech.wcomponents.util;
+
+import com.github.bordertech.wcomponents.file.File;
+import com.github.bordertech.wcomponents.file.FileItemWrap;
+import java.io.IOException;
+import java.text.DecimalFormat;
+import java.util.List;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.tika.Tika;
+
+/**
+ * Utility methods for {@link File}.
+ *
+ * @author Aswin Kandula
+ * @since 1.4
+ */
+public final class FileUtil {
+
+	/**
+	 * No instance methods here.
+	 */
+	private FileUtil() {
+	}
+	
+	/**
+	 * The logger instance for this class.
+	 */
+	private static final Log LOG = LogFactory.getLog(FileUtil.class);
+
+	/**
+	 * Checks if the file item is one among the supplied file types.
+	 * 
+	 * @param newFile the file to be checked
+	 * @param fileTypes allowed file types
+	 * @return {@code true} if file type is valid
+	 */
+	public static boolean validateFileType(final FileItemWrap newFile, final List<String> fileTypes) {
```

Comment:
Handle null arguments...
If newFile is null, return false
IllegalArgument if fileTypes is null

Reply:
If `fileTypes` in WFileWidget is set to `null` then [WFileWidget.getFileTypes()](https://github.com/kandula-aswin/wcomponents/blob/dc3cfa5bd9b33176afe1d332ae7a739d87476e55/wcomponents-core/src/main/java/com/github/bordertech/wcomponents/WFileWidget.java#L51) returns an empty list.
So should `FileUtil.validateFileType()` return `IllegalArgumentException` or `True` if fileTypes  argument is `null`?